### PR TITLE
:bug: fix `--stdin` CLI option

### PR DESCRIPTION
**Pull request by AucaCoyan**
## Description of changes

WIP

## Relevant Issues

#56

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -136,10 +138,23 @@ fn is_file_extension(file: &Path, extension: &str) -> bool {
...
-        use clap::CommandFactory;
         Cli::command().debug_assert();
     }
+
+    #[test]
+    fn pipe_stdin_to_cli() {
+        // TODO: create a file instead of reading one in the repo
+        let mut file = File::create("tests.txt").unwrap();
+        file.write_all(b"Hello, world!").unwrap();
+        let mut binding = AssertCommand::cargo_bin(env!("CARGO_PKG_NAME")).unwrap();
```
Comment (amtoine) [inline src/main.rs:155]: > as said on Discord

i think you need to run `cargo build` for `$env.CARGO_PKG_NAME` to exist for real on the filesystem
```bash
cargo clean
cargo test
```
vs
```bash
cargo clean
cargo build
cargo test
```
=> the former fails and the latter runs successfully :ok_hand:

Comment (AucaCoyan) [reply src/main.rs]: Thanks!

**Comment by AucaCoyan** [comment]
Thank you @amtoine ! With your help I could solve the issue 😄

**Review by amtoine — COMMENTED** [review]
the changes look ok to me, but i'm not sure they address the original issue :thinking: 

i'm running the following, on top of the PR branch and after a `cargo clean`:
- `cargo build --release` to create `./target/release/nufmt`
- `foo out> foo.txt` to create a dummy file

and then
1. `./target/release/nufmt --stdin (open foo.txt)` still works as before
2. `open foo.txt | ./target/release/nufmt` gives an error
3. `open foo.txt | ./target/release/nufmt --stdin` also gives an error

i expected 
- `1.` to not work anymore
- `2.` to give an error as it did
- `3.` to work as 1. does

**Comment by AucaCoyan** [comment]
Hello! I break the API and forgot to tell you 😆 . Thank you for listing every case. 
Reading it one more time today, I realized I made a mistake on the assignment. Let's see if I can fix it 😄

**Comment by AucaCoyan** [comment]
superseded by #61

**Pull request closed without merging**